traversal/scanner/GraphIterator.java

```diff
@@ -151,7 +151,12 @@ private VertexMap toVertexMap() {
         Map<Identifier.Variable.Retrievable, Vertex<?, ?>> answer = new HashMap<>();
         for (ProcedureVertex<?, ?> procedureVertex : procedure.vertices()) {
             if (procedureVertex.id().isRetrievable() && modifiers.filter().variables().contains(procedureVertex.id().asVariable().asRetrievable())) {
-                answer.put(procedureVertex.id().asVariable().asRetrievable(), vertexTraversers.get(procedureVertex).vertex());
+                Vertex<?, ?> vertex = vertexTraversers.get(procedureVertex).vertex();
+                if (vertex.isThing() && vertex.asThing().isAttribute() && vertex.asThing().asAttribute().isValue()) {
```

Comment:
same logic as in `VertexProcedure` - we convert `AttributeVertex.Value` back to `AttributeVertex` so that all layers of classes above the traversal receive concepts with types.